tests/end_to_end_tests/run_in_remote_machine/ec2.py

```diff
@@ -0,0 +1,370 @@
+            stderr = getattr(e.original_error, "stderr", None)
+            raise Exception(f"Deployment is not successful.\nStdout: {stdout}\nStderr: {stderr}")
+
+        return node
+
+    def run_command():
+
+        ssh = paramiko.SSHClient()
+        ssh.load_system_host_keys()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(
+            hostname=node.public_ips[0],
+            port=22,
+            username=ec2_image.ssh_username,
+            key_filename=str(private_key_path)
+        )
+
+        final_command = [
+            "/tmp/test_runner",
+            "-s",
+            *command
+        ]
+
+        command_str = shlex.join(final_command)
+        stdin, stdout, stderr = ssh.exec_command(
+            command=f"TEST_RUNS_REMOTELY=1 sudo -E {command_str} 2>&1",
+        )
+
+        print(f"stdout: {stdout.read().decode()}")
+
+        ssh.close()
+
+        assert stdout.channel.recv_exit_status() == 0, "Remote test execution has failed with."
+
+    file_mappings = file_mappings or {}
+    start_time = int(time.time())
+
+    driver_cls = get_driver(Provider.EC2)
+    driver = driver_cls(
+        access_key,
+        secret_key,
+        region=region
+    )
+
+    boto_client = boto3.client(
+        "ec2",
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key,
+        region_name=region
+    )
+
+    # Add current public IP to security group's prefix list.
+    # We have to update that prefix list each time because there are to many GitHub actions public IPs, and
+    # it is not possible to whitelist all of them in the AWS prefix list.
+    add_current_ip_to_prefix_list(
+        client=boto_client,
+        prefix_list_id=security_groups_prefix_list_id,
+        workflow_id=workflow_id
+    )
+
+    time.sleep(5)
+
+    node = None
+    try:
+        node = prepare_node()
+        run_command()
+    finally:
+        if node:
+            destroy_node_and_cleanup(driver=driver, node=node)
+
+    duration = int(time.time()) - start_time
+
+    print(f"Succeeded! Duration: {duration} seconds")
+
+def get_prefix_list_version(client, prefix_list_id: str):
+    """
+    Get version of the prefix list.
+    :param client: ec2 boto3 client.
+    :param prefix_list_id: ID of the prefix list.
+    """
+    resp = client.describe_managed_prefix_lists(
+        Filters=[
+            {
+                "Name": "prefix-list-id",
+                "Values": [prefix_list_id]
+            },
+        ],
+    )
+    found = resp["PrefixLists"]
+    assert len(found) == 1, f"Number of found prefix lists has to be 1, got {len(found)}"
+    prefix_list = found[0]
+    return int(prefix_list["Version"])
+
+
+def add_current_ip_to_prefix_list(
+    client,
+    prefix_list_id: str,
+    workflow_id: str = None
+):
+    """
+    Add new CIDR entry with current public IP in to the prefix list. We also additionally store json object in the
```

Comment:
Would be good to add a comment somewhere on why maintaining the custom prefix list is necessary.